rust/src/main.rs

```diff
@@ -50,65 +50,48 @@ where
     Ok(())
 }
 
-fn write_document_lengths(input: &Path, output_basename: &str) -> Result<u32> {
-    eprintln!("Processing document lengths...");
-    let input_file = File::open(input)?;
-
-    let num_documents = BufReader::new(&File::open(input)?)
-        .lines()
-        .count()
-        .to_u32()
-        .ok_or("Number of documents too large (must fit in u32)")?;
-
-    let input = BufReader::new(&input_file);
-    let mut sizes = BufWriter::new(File::create(format!("{}.sizes", output_basename))?);
-    let mut trecids = BufWriter::new(File::create(format!("{}.documents", output_basename))?);
-
-    let bar = ProgressBar::new(num_documents as u64);
-    bar.set_style(pb_style());
-    bar.set_draw_delta(num_documents as u64 / 100);
-    sizes.write_all(&num_documents.to_ne_bytes())?;
-    for (expected_docid, line) in input.lines().enumerate() {
-        let line = line?;
-        let mut columns = line.split('\t');
-        let docid: u32 = columns.next().ok_or("Corrupted file")?.parse()?;
-        let trecid = columns.next().ok_or("Corrupted file")?;
-        let length: u32 = columns.next().ok_or("Corrupted file")?.parse()?;
-        assert_eq!(
-            docid as usize, expected_docid,
-            "Document sizes must come in order"
-        );
-        sizes.write_all(&length.to_ne_bytes())?;
-        writeln!(trecids, "{}", trecid)?;
-        bar.inc(1);
-    }
-    bar.finish();
-    Ok(num_documents)
+fn print_header(hdr: Header) {
```

Comment:
Let's just call it what it is: `header`. Also, you can pass by reference so that you don't have to clone (see comment below):
```rust
fn print_header(header: &Header) {
```
Note that this is a immutable reference.